### PR TITLE
ROSA Shared VPC auto mode

### PR DESCRIPTION
**Pull request by yunjiang29**

### CODE REVIEW

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-shared-vpc-auto-e2e-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-shared-vpc-auto-e2e-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-shared-vpc-auto-e2e-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-shared-vpc-auto-e2e-f7

**Comment by yunjiang29** [comment]
test

https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/49854/rehearse-49854-periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-shared-vpc-auto-e2e-f7/1768530201999839232/artifacts/rosa-aws-sts-shared-vpc-auto-e2e-f7/rosa-cluster-provision/build-log.txt

```
Cluster ci-rosa-s-540b is being created with cluster-id: 2a0qbngdukskg5n6l1clf1c2rlt0b05v
Waiting for cluster ready...
Cluster state: waiting
Shared-VPC: Auto mode is enabled, adding ingress operator arn to shrared-role's trust policy
Shared-VPC: cluster status: Failed to verify ingress operator for shared VPC: Failed to assume role with ARN 'arn:aws:iam::6417***:role/ci-rosa-s-540b-shared-role': operation error STS: AssumeRole, https response error StatusCode: 403, RequestID: 272a20d8-904a-4e29-8246-1f57fe147b71, api error AccessDenied: User: arn:aws:sts::3017***:assumed-role/ci-rosa-s-540b-i7l8-openshift-ingress-operator-cloud-credentials/OCM is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::6417***:role/ci-rosa-s-540b-shared-role
Shared-VPC: Status match, waiting for 2 mins to make sure operator role is ready.
Shared-VPC: ingress: arn:aws:iam::3017***:role/ci-rosa-s-540b-i7l8-openshift-ingress-operator-cloud-credentials, shared_role: ci-rosa-s-540b-shared-role, installer: arn:aws:iam::3017***:role/ci-op-r2iw0ngw-Installer-Role
Shared-VPC: Applied new policy.
Shared-VPC: waiting for 2 mins to make sure the cluster status is up to date.
Cluster state: validating
Cluster state: validating
Cluster state: pending
Cluster state: installing
Cluster state: installing
```

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-rosa_hcp-4.15-nightly-x86-control-plane-3nodes

**Comment by yunjiang29** [comment]
/pj-rehearse pull-ci-openshift-aws-load-balancer-operator-main-e2e-aws-rosa-operator

File: ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh [hunk 1]
```diff
@@ -536,8 +551,9 @@ rosa create cluster -y \
                     ${SHARED_VPC_SWITCH} \
                     ${SECURITY_GROUP_ID_SWITCH} \
                     ${NO_CNI_SWITCH} \
-                    ${DRY_RUN_SWITCH} \
-                    | sed "s/$AWS_ACCOUNT_ID/$AWS_ACCOUNT_ID_MASK/g" > "${CLUSTER_INFO}"
+                    ${DRY_RUN_SWITCH} > "${CLUSTER_INFO_WITHOUT_MASK}"
+
+cat ${CLUSTER_INFO_WITHOUT_MASK} | sed "s/$AWS_ACCOUNT_ID/$AWS_ACCOUNT_ID_MASK/g" > "${CLUSTER_INFO}"
```
Comment (yasun1) [inline ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh:556]: It does not mask `SHARED_VPC_AWS_ACCOUNT_ID`, right?

**Comment by mgahagan73** [comment]
/lgtm

I don't think any of the test case failures have anything to do with this PR.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-advanced-f7

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-stable-aws-rosa-sts-advanced-f7

**Comment by yunjiang29** [comment]
@liangxia PTAL

**Pull request merged**